packages/lodestar/src/chain/chain.ts

```diff
@@ -165,10 +165,18 @@ export class BeaconChain extends EventEmitter implements IBeaconChain {
     const preFinalizedEpoch = state.finalizedEpoch;
     const preJustifiedEpoch = state.currentJustifiedEpoch;
     // Run the state transition
-    const newState = stateTransition(this.config, state, block, true);
+    let newState: BeaconState;
+    try {
+      newState = stateTransition(this.config, state, block, true);
+    }catch (e) {
+      // store block root in db and terminate
+      const blockRoot = hashTreeRoot(block, this.config.types.BeaconBlock);
+      await this.db.setBadBlockRoot(blockRoot);
+      this.logger.warn( "bad block" + e.message + '\n' + e.stack);
```

Comment:
Put as second line of catch statement, add blockroot to log, missing whitespace after "bad block", d o n' t p u t stack  th ere, you could p ri n t  stac k  asa  d ebu glog or something